SPHINXsys/src/shared/particles/particle_adaptation.cpp

```diff
@@ -34,8 +35,8 @@ namespace SPH
 	{
 		sph_body_ = sph_body;
 		system_domain_bounds_ = sph_body_->getSPHSystem().system_domain_bounds_;
-		Real system_resolution = sph_body_->getSPHSystem().resolution_ref_;
-		spacing_ref_ = RefinedSpacing(system_resolution, global_refinement_level_);
+		Real body_resololution = system_resolution_ratio_ * sph_body_->getSPHSystem().resolution_ref_;
+		spacing_ref_ = RefinedSpacing(body_resololution, global_refinement_level_);
```

Comment:
@Xiangyu-Hu here I added an optional scaling factor, it can be changed by SetSystemResolutionRatio